pkgs/by-name/vl/vlc-bittorrent/package.nix

```diff
@@ -0,0 +1,66 @@
+{ autoconf
+, autoconf-archive
+, automake
+, boost
+, fetchFromGitHub
+, gcc
+, gnumake
+, lib
+, libtool
+, libtorrent-rasterbar
+, libvlc
+, openssl
+, pkg-config
+, stdenv
+}:
+
+# VLC does not know where the vlc-bittorrent package is installed.
+# make sure to have something like:
+#   environment.variables.VLC_PLUGIN_PATH = "${pkgs.vlc-bittorent}";
+
+stdenv.mkDerivation rec {
+  pname = "vlc-bittorrent";
+  version = "2.15.0";
+
+  src = fetchFromGitHub {
+    owner = "johang";
+    repo = pname;
+    rev = "6810d479e6c1f64046d3b30efe78774b49d1c95b";
+    hash = "sha256-7FHeQYHbMKZJ3yeHqxTTAUwghTje+gEX8gSEJzfG5sQ=";
+  };
+
+  nativeBuildInputs = [
+    gcc
+    autoconf
+    autoconf-archive
+    automake
+    gnumake
+    pkg-config
+  ];
+
+  buildInputs = [
+    libtorrent-rasterbar
+    libvlc
+    libtool
+    openssl
+    boost
+  ];
+
+  buildPhase = ''
+    autoreconf -i
+    ./configure --prefix=$out
+    make
+  '';
+
+  installPhase = ''
+    make install
+  '';
+
+  meta = with lib; {
+    description = "A bittorrent plugin for VLC.";
+    homepage = "https://github.com/johang/vlc-bittorrent";
+    license = licenses.gpl3;
+    platforms = platforms.all;
```

Comment:
If it uses Autotools, it's _probably_ (at least...) UNIX only, please make sure this is correct.

Reply:
> ### Does it work on Windows, Mac OS X, Android, Windows RT, iOS, my toaster?
> Probably. I have not tested. It should run since libtorrent works on most systems and the plugin is just standard C and C++. Patches are welcome.

[Quote from upstream](https://github.com/johang/vlc-bittorrent?tab=readme-ov-file#does-it-work-on-windows-mac-os-x-android-windows-rt-ios-my-toaster) does not specifically say it works on all platforms. It only *states* it works on Linux. So I assume it should be linux only?

Reply:
It should be fine to start with, if somebody on Darwin ends up wanting to use it (and it works), they can make a PR to add it to the platforms.